### PR TITLE
HDDS-3170. Fix issues in File count by size task.

### PR DESCRIPTION
**Pull request by avijayanhwx**
## What changes were proposed in this pull request?
Recon has a task to maintain the number of files created in Ozone across various size bins (https://issues.apache.org/jira/browse/HDDS-1366). However, there are some bugs in the task's handling of DELETE and existing key PUT key operations from OM. This patch fixes the issues and adds tests to capture issues. 

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-3170

## How was this patch tested?
Manually tested on docker.
Added unit tests for the new cases.

### CODE REVIEW

**Comment by avijayanhwx** [comment]
@vivekratnavel / @shwetayakkali / @swagle Please review.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OMDBUpdatesHandler.java [hunk 1]
```diff
@@ -88,30 +93,47 @@ private void processEvent(int cfIndex, byte[] keyBytes, byte[]
...
           new OMDBUpdateEvent.OMUpdateEventBuilder<>();
       builder.setTable(tableName);
+      builder.setAction(action);
 
-      Object key = codecRegistry.asObject(keyBytes, keyType);
+      String key = codecRegistry.asObject(keyBytes, keyType);
       builder.setKey(key);
 
-      if (!action.equals(OMDBUpdateEvent.OMDBUpdateAction.DELETE)) {
+      if (action.equals(PUT)) {
```
Comment (swagle) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OMDBUpdatesHandler.java]: Minor: Can be changed to == enum compare instead.

File: hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java [hunk 2]
```diff
@@ -229,22 +228,39 @@ void populateFileCountBySizeDB() {
...
+          "found in Recon OM DB : {}", key);
+    } else {
+      int binIndex = calculateBinIndex(omKeyInfo.getDataSize());
+      if (upperBoundCount[binIndex] > 0) {
         //decrement only if it had files before, default DB value is 0
         upperBoundCount[binIndex]--;
       } else {
         LOG.warn("Unexpected error while updating bin count. Found 0 count " +
-            "for index : {} while processing DELETE event for {}", binIndex,
+                "for index : {} while processing DELETE event for {}", binIndex,
```
Comment (swagle) [inline hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTask.java]: Minor: bad indent

**Comment by vivekratnavel** [comment]
+1 LGTM

**Comment by avijayanhwx** [comment]
Thank you for the reviews @swagle , @vivekratnavel & @adoroszlai.

**Pull request merged**